### PR TITLE
APPSEC-1056 S6336(secrets): Improvements to raise less FPs

### PR DESCRIPTION
**Pull request by daniel-teuchert-sonarsource**

### CODE REVIEW

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/alibaba.yaml [hunk 1]
```diff
@@ -64,30 +69,29 @@ provider:
...
       detection:
+        pre:
+          include:
+            content:
+              - aliyun
+              - alibaba
+              - alicloud
         matching:
-          pattern: "(?i)(?<![A-Z])ali(?:yun|baba|cloud).{0,50}['\"`]([0-9a-z]{30})['\"`]"
+          pattern: "(?i)(?:SECRET_?(?:ACCESS)?_?KEY|(?:ACCESS)?_?KEY_?SECRET)\\b[^0-9a-z]{0,10}([0-9a-z]{30})(?![a-z0-9\\/+=$\\-_])"
```
Comment (loris-s-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/alibaba.yaml:83]: What is the why for this change ? you removed pretty important prefixes here

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/alibaba.yaml [hunk 2]
```diff
@@ -64,30 +69,29 @@ provider:
...
           containsSecret: true
           match: l0GdwcDYdJwB1VJ5pv0ormyTV9nhvW
-      detection:
-        matching:
-          pattern: "(?i)(?:SECRET_?(?:ACCESS)?_?KEY|(?:ACCESS)?_?KEY_?SECRET)\\b[^0-9a-z]{0,10}([0-9a-z]{30})(?![a-z0-9\\/+=$\\-_])"
-        post:
-          statisticalFilter:
-            threshold: 4.2
-          patternNot:
-            - "EXAMPLE"
```
Comment (loris-s-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/alibaba.yaml:93]: I think you mixed up the detection patterns between two different detection logics

**Review by loris-s-sonarsource — CHANGES_REQUESTED** [review]

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/alibaba.yaml (hunk 2 above)
Comment (daniel-teuchert-sonarsource) [reply sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/alibaba.yaml:93]: I am pretty sure that this is only a confusing diff display.
Basically we have one pattern with the ali.. prefix and one with the SECRET... prefix.
I added pre filters to both to decrease the false positives.

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/alibaba.yaml (hunk 1 above)
Comment (daniel-teuchert-sonarsource) [reply sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/alibaba.yaml:83]: No, this is only a weird diff display. The structure of the file changed a bit. The actual changes to the regex are minor.

Comment (loris-s-sonarsource) [reply sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/alibaba.yaml:83]: Ah ok I see

**Review by loris-s-sonarsource — APPROVED** [review]
LGTM ! That diff's really weird

**Pull request merged**